File: libs/grpc/clients/events.go
package clients

import (
	"log"

	"github.com/twirapp/twir/libs/grpc/constants"
	"github.com/twirapp/twir/libs/grpc/events"

	"google.golang.org/grpc"
)

func NewEvents(env string) events.EventsClient {
	serverAddress := createClientAddr(env, "events", constants.EVENTS_SERVER_PORT)

	conn, err := grpc.Dial(
		serverAddress,
		defaultClientsOptions...,
	)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	c := events.NewEventsClient(conn)
	return c
}

File: libs/grpc/clients/events.ts
import { ChannelCredentials, createChannel, createClient } from 'nice-grpc'

import { CLIENT_OPTIONS, createClientAddr, waitReady } from './helper.js'
import { PORTS } from '../constants/constants.js'
import { EventsDefinition } from '../events/events'

import type { EventsClient } from '../events/events'

export async function createEvents(env: string): Promise<EventsClient> {
	const channel = createChannel(
		createClientAddr(env, 'events', PORTS.EVENTS_SERVER_PORT),
		ChannelCredentials.createInsecure(),
		CLIENT_OPTIONS,
	)

	await waitReady(channel)

	const client = createClient(EventsDefinition, channel)

	return client as any
}
